docs/DeveloperGuide.md

```diff
@@ -1,26 +1,237 @@
+place is not found, it will create a location and allocate it to be OutOfNUS.  
+
+This feature also allows users to search for a location based on a given location name, or the location where an event 
+is held. It will print out additional information about the location to the user. 
+
+The `locate` command executes with the following steps: 
+  
+Step 1. When the user type in `locate [input]`, the Parser class will extract the input from the command and call a 
+`LocateCommand` object.  
+* `input` will always be a String when passed into the command object. 
+
+Step 2. `LocateCommand` will then try to convert the input into an integer, if possible.  
+
+Step 3. If conversion is possible, the method will obtain the event at the given index and return its location 
+information.  
+* Tries to catch `IndexOutOfBoundException` to check if the given index is within the range of the
+event list.
+
+Step 4. If conversion is not possible, then it will check if the given string is a valid location.  
+* Returns the location if valid
+* Prints an error message if invalid   
+
+The sequence diagram below shows exactly which methods, from which classes, are called to obtain the required location. 
+![](diagrams/Locate.jpg)
+*Figure 4.4.1 Sequence diagram for locate function*
+
+### 4.5 Reminders
+
+### 4.6 Sort events
+
+### 4.7 View events
+
+## 5. Documentation
+
+### 5.1 Product scope
+__Target user profile:__
+* NUS student or staff
+* able to type quickly
+* has quite a number of events to keep track of  
+* prefers to use desktop apps
+* prefers using Command Line Interface (CLI) apps
+* prefers typing instead of mouse interactions
+
+### 5.2 Value proposition
 
 {Describe the value proposition: what problem does it solve?}
 
-## User Stories
+### 5.3 User Stories
 
 |Version| As a ... | I want to ... | So that I can ...|
 |--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
-
+|1.0|busy individual|keep track of both personal and school activities|avoid clashing events
+|1.0|freshman|know module details and lesson venues|be on time for my activities
+|1.0|forgetful person|be reminded of the deadlines for my assignments|submit on time
+|1.0|tutor|know the estimated time for my students to travel to their next class|pace my lesson suitably
+|1.0|hardworking student|track how much time I have spent studying\allocate my time efficiently
+|2.0|exchange student|know the optimal path to reach my next destination|avoid getting lost
+|2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+|2.0|professor|Know whether my students have another lesson after mine and the expected time of travelling|Pace my lesson appropriately
```

Comment:
Table format is broken

![image](https://user-images.githubusercontent.com/8433829/97530464-27c38280-19ed-11eb-8f96-f50888248c64.png)